crates/ev-enclave/src/version/mod.rs

```diff
@@ -34,18 +34,36 @@ impl CliError for VersionError {
     }
 }
 
-pub async fn get_runtime_and_installer_version(
-    from_existing: Option<String>,
-) -> Result<(String, String), VersionError> {
-    match from_existing {
-        Some(existing) => parse_version_from_existing_dockerfile(existing),
-        None => {
-            let enclave_build_assets_client = EnclaveAssetsClient::new();
-            let data_plane_version = enclave_build_assets_client.get_data_plane_version().await?;
-            let installer_version = enclave_build_assets_client
-                .get_runtime_installer_version()
-                .await?;
-            Ok((data_plane_version, installer_version))
+pub struct EnclaveRuntime {
+    pub data_plane_version: String,
+    pub installer_version: String,
+}
+
+impl EnclaveRuntime {
+    pub async fn new() -> Result<EnclaveRuntime, VersionError> {
```

Comment:
this constructor seems odd, would it not make more sense to have:

```rust
impl std::convert::From<RuntimeMajorVersion> for EnclaveRuntime {
  fn from(value: RuntimeMajorVersion) -> Self {
    Self {
      data_plane_version: value.latest,
      installer_version: versions.installer
    }
  }
}
```

and replace uses of EnclaveRuntime::new().await? with:
```rust
let versions: EnclaveRuntime = client.get_runtime_versions().await?.into();
```

Reply:
This is actually really hard to deal with since the commands return exitcodes not results. So you `?` the error.

You can do it this way but when you want to match on an existing docker file you end up with something quite verbose along the lines of this. (Won't compile)
```rust
    let try_enclave_runtime = match &build_args.from_existing {
        Some(dockerfile) => parse_version_from_existing_dockerfile(dockerfile),
        None => {
            let enclave_assets_client = EnclaveAssetsClient::new();
            enclave_assets_client
                .get_runtime_versions()
                .await
                .map_err(VersionError::ApiError)
                .map(|versions| versions.into())
        }
    };

    let enclave_runtime = match try_enclave_runtime {
        Ok(runtime) => runtime,
        Err(e) => {
            log::error!("Failed to retrieve the latest data plane and installer versions - {e:?}");
            return e.exitcode();
        }
    };
```